opensearch/src/main/java/org/opensearch/sql/opensearch/storage/scan/OpenSearchPagedIndexScanBuilder.java

```diff
@@ -22,6 +24,18 @@ public OpenSearchPagedIndexScanBuilder(OpenSearchPagedIndexScan indexScan) {
     this.indexScan = indexScan;
   }
 
+  @Override
+  public boolean pushDownPagination(LogicalPaginate paginate) {
+    var builder = indexScan.getRequestBuilder();
+    if (builder instanceof InitialPageRequestBuilder) {
+      builder.pushDownPageSize(paginate.getPageSize());
+      return false;
```

Comment:
If I return `true`, `PaginateOperator` is being removed from the tree alongs with `ProjectOperator`.